Scripts/build/Jenkins/Jenkinsfile

```diff
@@ -0,0 +1,665 @@
+#!/usr/bin/env groovy
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+PIPELINE_CONFIG_FILE = 'scripts/build/Jenkins/lumberyard.json'
+INCREMENTAL_BUILD_SCRIPT_PATH = 'scripts/build/bootstrap/incremental_build_util.py'
+
+EMPTY_JSON = readJSON text: '{}'
+
+PROJECT_REPOSITORY_NAME = 'o3de-netsoaktest'
+PROJECT_ORGANIZATION_NAME = 'o3de'
+ENGINE_REPOSITORY_NAME = 'o3de'
+ENGINE_ORGANIZATION_NAME = 'o3de'
+ENGINE_BRANCH_DEFAULT = "${env.BRANCH_DEFAULT}" ?: "${env.BRANCH_NAME}"
+
+def pipelineProperties = []
+
+def pipelineParameters = [
+    // Build/clean Parameters
+    // The CLEAN_OUTPUT_DIRECTORY is used by ci_build scripts. Creating the parameter here passes it as an environment variable to jobs and is consumed that way
+    booleanParam(defaultValue: false, description: 'Deletes the contents of the output directory before building. This will cause a \"clean\" build. NOTE: does not imply CLEAN_ASSETS', name: 'CLEAN_OUTPUT_DIRECTORY'),
+    booleanParam(defaultValue: false, description: 'Deletes the contents of the output directories of the AssetProcessor before building.', name: 'CLEAN_ASSETS'),
+    booleanParam(defaultValue: false, description: 'Deletes the contents of the workspace and forces a complete pull.', name: 'CLEAN_WORKSPACE'),
+    booleanParam(defaultValue: false, description: 'Recreates the volume used for the workspace. The volume will be created out of a snapshot taken from main.', name: 'RECREATE_VOLUME'),
+    stringParam(defaultValue: "${ENGINE_BRANCH_DEFAULT}", description: 'Sets a different branch from o3de engine repo to use or use commit id. Default is branchname', trim: true, name: 'ENGINE_BRANCH')
+]
+
+def palSh(cmd, lbl = '', winSlashReplacement = true) {
+    if (env.IS_UNIX) {
+        sh label: lbl,
+           script: cmd
+    } else if (winSlashReplacement) {
+        bat label: lbl,
+            script: cmd.replace('/','\\')
+    } else {
+        bat label: lbl,
+            script: cmd
+    }
+}
+
+def palMkdir(path) {
+    if (env.IS_UNIX) {
+        sh label: "Making directories ${path}",
+           script: "mkdir -p ${path}"
+    } else {
+        def win_path = path.replace('/','\\')
+        bat label: "Making directories ${win_path}",
+            script: "mkdir ${win_path}."
+    }
+}
+
+def palRm(path) {
+    if (env.IS_UNIX) {
+        sh label: "Removing ${path}",
+           script: "rm ${path}"
+    } else {
+        def win_path = path.replace('/','\\')
+        bat label: "Removing ${win_path}",
+            script: "del /Q ${win_path}"
+    }
+}
+
+def palRmDir(path) {

```

Comment:
nit: Case convention is not consistent with other functions in file. Not sure what pal refers to here, assume its shorthand for platform, maybe consider not abbreviating name?